packages/material-ui/src/ClickAwayListener/ClickAwayListener.js

```diff
@@ -16,10 +16,17 @@ function mapEventPropToEvent(eventProp) {
  * For instance, if you need to hide a menu when people click anywhere else on your page.
  */
 const ClickAwayListener = React.forwardRef(function ClickAwayListener(props, ref) {
```

Comment:
I wonder about removing the forward ref logic for all the components that would forward it to the underlying DOM element, so far it includes ClickAwayLister, and Tooltip. The motivation would be that there is no underlying new DOM node anyway. cc @eps1lon.

Reply:
As long as you still keep `cloneElement` to avoid someone stealing the ref from us by accident :) But any removal could be considered a breaking change. Maybe for v5?

Reply:
The forward ref logic for these two components was introduced in disguised pull requests. I never came from user feedback but from me noticing the inconsistency.

Reply:
Oh, actually, I think that we should remove it here. The ref forwarding is inconsistent with the usage of exactProp. I believe we should have ref and props forward identically.

Reply:
I guess we can remove and revert if someone complains (very small chance)

Reply:
> I believe we should have ref and props forward identically.

So, in this configuration, I think that the usage of forward ref in the Tooltip is justified.

Reply:
For any future readers

> > I believe we should have ref and props forward identically.
> 
> So, in this configuration, I think that the usage of forward ref in the Tooltip is justified.

@oliviertassinari clarified this when I asked him via Twitter DM:

> => ref and props do no forward the same way
> ref goes on the wrapped DOM node, props don't.